src/bindings/PyDP/mechanisms/mechanism.cpp

```diff
@@ -1 +1,165 @@
-#include "mechanism.h"
\ No newline at end of file
+#include <pybind11/pybind11.h>
+#include "pybind11/complex.h"
+#include "pybind11/functional.h"
+#include "pybind11/stl.h"
+
+#include "algorithms/numerical-mechanisms.h"
+
+#include "../pydp_lib/algorithm_builder.hpp"
+
+namespace py = pybind11;
+namespace dp = differential_privacy;
+
+class ConfidenceIntervalBinder {
+ public:
+  static void DeclareIn(py::module& m) {
+    py::class_<dp::ConfidenceInterval> confidence_interval(m, "ConfidenceInterval");
+    confidence_interval.attr("__module__") = "pydp";
+    confidence_interval
+        .def_property("lower_bound", &dp::ConfidenceInterval::lower_bound,
+                      &dp::ConfidenceInterval::set_lower_bound)
+        .def_property("upper_bound", &dp::ConfidenceInterval::upper_bound,
+                      &dp::ConfidenceInterval::set_upper_bound)
+        .def_property("confidence_level", &dp::ConfidenceInterval::confidence_level,
+                      &dp::ConfidenceInterval::set_confidence_level);
+  }
+};
+
+template <typename T>
+py::class_<dp::NumericalMechanism>& DefPyAddNoise(
+    py::class_<dp::NumericalMechanism>& pyclass) {
+  using FunctorType = T (dp::NumericalMechanism::*)(T);
+  return pyclass.def("add_noise",
+                     static_cast<FunctorType>(&dp::NumericalMechanism::AddNoise),
+                     py::arg("result"));
+}
+
+template <typename T, typename U>
+py::class_<dp::NumericalMechanism>& DefPyAddNoise(
+    py::class_<dp::NumericalMechanism>& pyclass) {
+  using FunctorType = T (dp::NumericalMechanism::*)(T, U);
+  return pyclass.def("add_noise",
+                     static_cast<FunctorType>(&dp::NumericalMechanism::AddNoise),
+                     py::arg("result"), py::arg("privacy_budget"));
+}
+
+template <typename T, typename U>
```

Comment:
can we add comments to explain what this function does?

Reply:
@chinmayshah99 i'm not really sure what this does also since I haven't touched any of the C++ code